### ISSUE TITLE
Wrong delimiter for cross-platform multiplayer.

### ISSUE THREAD

**Issue by o01eg** [freeorion/freeorion#2205]
### Environment
* **FreeOrion Version**: 2018-06-24.4983374 0.4.8 pre-RC2
* **Operating System**: Ubuntu Linux 18.04.0 x64
* **Fetched as** 
  * Compiled from source

### Description
I'm not sure but it looks like UI for loading save game use local delimiter not remote one. This cause wrong filename sent to the server. These are logs from multiplayer server. I suppose `Rudy` at 18:03 use Windows. I use Linux at 18:33:

```
18:03:28.630091 [debug] server : SaveGamePreviewUtils.cpp:93 : Loaded preview with: 2 human players
18:03:28.630104 [debug] server : SaveGamePreviewUtils.cpp:101 : LoadSaveGamePreviewData: Successfully loaded preview from: /var/lib/freeorion/save/./MeRuSeSp212/FreeOrion_0056_20180705_171854.mps
18:03:28.630136 [debug] server : ServerApp.cpp:971 : ServerApp::UpdateSavePreviews: Sending 87 previews in response.
18:03:30.124056 [debug] FSM : ServerFSM.cpp:977 : Get message from host or allowed player.
18:03:30.124115 [debug] FSM : ServerFSM.cpp:104 : PlayerSetupData:
18:03:30.124134 [debug] FSM : ServerFSM.cpp:119 :  ... -1 : AI_1, AI_PLAYER, SP_CHATO
18:03:30.124145 [debug] FSM : ServerFSM.cpp:119 :  ... -1 : AI_2, AI_PLAYER, SP_ETTY
18:03:30.124155 [debug] FSM : ServerFSM.cpp:119 :  ... 1 : Rudy, PLAYER, SP_EAXAW
18:03:30.124234 [debug] server : SaveLoad.cpp:542 : LoadEmpireSaveGameData: filename: /var/lib/freeorion/save/./MeRuSeSp212\FreeOrion_0088_20180705_174533.mps path:"/var/lib/freeorion/save/./MeRuSeSp212\FreeOrion_0088_20180705_174533.mps"
18:03:30.124284 [error] server : SaveLoad.cpp:574 : Error when reading save file. LoadEmpireSaveGameData exception: : Unable to open file
...
18:33:54.692762 [debug] server : SaveGamePreviewUtils.cpp:66 : LoadSaveGamePreviewData: Loading preview from: /var/lib/freeorion/save/./MeRuSeSp212/FreeOrion_
0056_20180705_171854.mps
18:33:54.692803 [debug] server : SaveGamePreviewUtils.cpp:83 : Deserializing XML data
18:33:54.692927 [debug] server : SaveGamePreviewUtils.cpp:93 : Loaded preview with: 2 human players
18:33:54.692936 [debug] server : SaveGamePreviewUtils.cpp:101 : LoadSaveGamePreviewData: Successfully loaded preview from: /var/lib/freeorion/save/./MeRuSeSp2
12/FreeOrion_0056_20180705_171854.mps
18:33:54.692958 [debug] server : ServerApp.cpp:971 : ServerApp::UpdateSavePreviews: Sending 87 previews in response.
18:34:05.090432 [debug] FSM : ServerFSM.cpp:977 : Get message from host or allowed player.
18:34:05.090502 [debug] FSM : ServerFSM.cpp:104 : PlayerSetupData:
18:34:05.090523 [debug] FSM : ServerFSM.cpp:119 :  ... -1 : AI_1, AI_PLAYER, SP_CHATO
18:34:05.090536 [debug] FSM : ServerFSM.cpp:119 :  ... -1 : AI_2, AI_PLAYER, SP_ETTY
18:34:05.090550 [debug] FSM : ServerFSM.cpp:119 :  ... 1 : Rudy, PLAYER, SP_EAXAW
18:34:05.090562 [debug] FSM : ServerFSM.cpp:119 :  ... 2 : Rudy2, PLAYER, SP_GEORGE
18:34:05.090574 [debug] FSM : ServerFSM.cpp:119 :  ... 3 : o01eg, PLAYER, SP_LAENFA
18:34:05.090650 [debug] server : SaveLoad.cpp:542 : LoadEmpireSaveGameData: filename: /var/lib/freeorion/save/./MeRuSeSp212/FreeOrion_0088_20180705_174533.mps
 path:"/var/lib/freeorion/save/./MeRuSeSp212/FreeOrion_0088_20180705_174533.mps"
```

### Expected Result
Successful load game.

### Steps to reproduce
* Launch server on Linux.
* Start game, save into subfolder.
* Try to load game from subfolder from Windows client.

**Comment by geoffthemedio** [comment]
Try with that pull request applied?

**Comment by o01eg** [comment]
I've published PR on multiplayer server. I'm going to look in logs if it will be fixed.

**Comment by geoffthemedio** [comment]
Probably need another approach ( https://github.com/freeorion/freeorion/pull/2206#discussion_r200492276 ) but knowing if that change works is also helpful.

**Comment by o01eg** [comment]
First approach with patching `SaveLoad.cpp` didn't solve issue:
```
20:22:07.220395 [debug] server : SaveGamePreviewUtils.cpp:101 : LoadSaveGamePrev
iewData: Successfully loaded preview from: /var/lib/freeorion/save/./CoHiRuTrVu9
77/FreeOrion_0237_20180701_225813.mps
20:22:07.220416 [debug] server : ServerApp.cpp:971 : ServerApp::UpdateSavePrevie
ws: Sending 261 previews in response.
20:22:14.848221 [debug] FSM : ServerFSM.cpp:977 : Get message from host or allow
ed player.
20:22:14.848275 [debug] FSM : ServerFSM.cpp:104 : PlayerSetupData:
20:22:14.848294 [debug] FSM : ServerFSM.cpp:119 :  ... 1 : Rudy, PLAYER, SP_SCYL
IOR
20:22:14.848383 [debug] server : SaveLoad.cpp:542 : LoadEmpireSaveGameData: file
name: /var/lib/freeorion/save/./CoHiRuTrVu977\FreeOrion_0262_20180701_233942.mps
 path:"/var/lib/freeorion/save/./CoHiRuTrVu977\FreeOrion_0262_20180701_233942.mp
s"
20:22:14.848625 [error] server : SaveLoad.cpp:574 : Error when reading save file
. LoadEmpireSaveGameData exception: : Unable to open file
```

**Comment by Vezzra** [comment]
@o01eg, that's strange, because when I apply #2206 to the release branch and build that on Windows, I can load savegames on your server, [see my comment here](https://github.com/freeorion/freeorion/pull/2206#issuecomment-403289001).

**Comment by dbenage-cx** [comment]
The log above is from applying a previous version of PR which modified `LoadEmpireSaveGameData` and not Directories.cpp (IIRC)

**Comment by Vezzra** [comment]
Ah ok, didn't catch that.

**Comment by o01eg** [comment]
Good, looks like it fixed.

**Comment by Vezzra** [comment]
Fixed by #2206.
